integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+        super();
+    }
+
+
+    /*
+     * Instance methods.
+     */
+
+
+    /**
+     * Creates and returns a non-{@code null} {@link ConfigSource} that sources its values from an Oracle Cloud
+     * Infrastructure (OCI) <a
+     * href="https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Concepts/keyoverview.htm">Vault</a>.
+     *
+     * @param type one of the {@linkplain #supported() supported types}, or an {@linkplain ConfigSources#empty() empty
+     * <code>ConfigSource</code>} will be returned
+     *
+     * @param metaConfig a {@link Config} serving as meta-configuration for this provider; must not be {@code null} when
+     * {@code type} is {@linkplain #supports(String) supported}
+     *
+     * @return a non-{@code null} {@link ConfigSource}
+     *
+     * @exception NullPointerException if {@code type} is {@linkplain #supports(String) supported} and {@code
+     * metaConfig} is {@code null}
+     *
+     * @see #supported()
+     *
+     * @deprecated For use by the Helidon Config subsystem only.
+     */
+    @Deprecated // For use by the Helidon Config subsystem only.
+    @Override // ConfigSourceProvider
+    public ConfigSource create(String type, Config metaConfig) {
+        return this.supports(type) ? SecretBundleConfigSource.builder().config(metaConfig).build() : ConfigSources.empty();
+    }
+
+    /**
+     * Returns a non-{@code null}, immutable {@link Set} of supported types suitable for the Helidon Config subsystem to
+     * pass to the {@link #create(String, Config)} method.
+     *
+     * <p>This method returns a {@link Set} whose sole element is the string "{@code oci-secrets}".</p>
+     *
+     * @return a non-{@code null}, immutable {@link Set}
+     *
+     * @see #create(String, Config)
+     *
+     * @deprecated For use by the Helidon Config subsystem only.
+     */
+    @Deprecated // For use by the Helidon Config subsystem only.
+    @Override // ConfigSourceProvider
+    public Set<String> supported() {
+        return SUPPORTED_TYPES;
+    }
+
+    /**
+     * Returns {@code true} if and only if the supplied {@code type} is non-{@code null} and the {@link Set} returned by
+     * an invocation of the {@link #supported()} method {@linkplain Set#contains(Object) contains} it.
+     *
+     * @param type the type to test; may be {@code null} in which case {@code false} will be returned
+     *
+     * @return {@code true} if and only if the supplied {@code type} is non-{@code null} and the {@link Set} returned by
+     * an invocation of the {@link #supported()} method {@linkplain Set#contains(Object) contains} it
+     *
+     * @see #supported()
+     *
+     * @see #create(String, Config)
+     *
+     * @deprecated For use by the Helidon Config subsystem only.
+     */
+    @Deprecated // For use by the Helidon Config subsystem only.
+    @Override // ConfigSourceProvider
+    public boolean supports(String type) {
+        return type != null && this.supported().contains(type);
```

Comment:
type cannot `null`, as we follow our own guidelines
we do not want to be be null safe - the exact opposite, we expect any component to fail with a NPE if `null` is sent over public API.

Reply:
I will eliminate the `null` check.

Reply:
Done